elements/pl-order-blocks/dag_checker_test.py

```diff
@@ -77,13 +78,23 @@ def test_grade_dag():
 problem_3_invalid_dag_2 = {'1': [], '2': [], '3': ['2'], '4': [], '5': ['4'], '6': ['3', '5']}
 problem_3_invalid_dag_3 = {'1': [], '2': [], '3': ['2'], '4': [], '5': ['4', 'g1'], '6': ['g1', 'g2']}
 problem_3_dag = {'1': [], '2': [], '3': ['2'], '4': [], '5': ['4'], '6': ['g1', 'g2'], 'g1': [], 'g2': []}
-problem_3_groups = {'1': None, '2': 'g1', '3': 'g1', '4': 'g2', '5': 'g2'}
+problem_3_groups = {'1': None, '2': 'g1', '3': 'g1', '4': 'g2', '5': 'g2', '6': None}
 
 
 def test_problem_validation():
-    assert validate_grouping(dag_to_nx(problem_1_dag), problem_1_groups)
-    assert validate_grouping(dag_to_nx(problem_2_dag), problem_2_groups)
-    assert not validate_grouping(dag_to_nx(problem_3_invalid_dag_1), problem_3_groups)
-    assert not validate_grouping(dag_to_nx(problem_3_invalid_dag_2), problem_3_groups)
-    assert not validate_grouping(dag_to_nx(problem_3_invalid_dag_3), problem_3_groups)
-    assert validate_grouping(dag_to_nx(problem_3_dag), problem_3_groups)
+    dag_to_nx(problem_1_dag, problem_1_groups)
+    dag_to_nx(problem_2_dag, problem_2_groups)
+    with pytest.raises(Exception):
+        dag_to_nx(problem_3_invalid_dag_1, problem_3_groups)
+    with pytest.raises(Exception):
+        dag_to_nx(problem_3_invalid_dag_2, problem_3_groups)
+    with pytest.raises(Exception):
+        dag_to_nx(problem_3_invalid_dag_3, problem_3_groups)
+    dag_to_nx(problem_3_dag, problem_3_groups)
+
+
+def test_solve_dag():
+    problems = [(problem_1_dag, problem_1_groups), (problem_2_dag, problem_2_groups), (problem_3_dag, problem_3_groups)]
+    for depends_graph, group_belonging in problems:
+        solution = solve_dag(depends_graph, group_belonging)
+        assert len(solution) == grade_dag(solution, depends_graph, group_belonging)[0]
```

Comment:
I prefer to leave it as is, there's actually a good reason for it.

If we were to hard code a particular solution here, then our tests passing would depend on the implementation details of `nx.topological_sort`, which in theory could return _any_ topological sort of the graph. It would seem weird if we had to update our tests because an implementation detail in a library changed. It would also seem weird to make it look like a particular solution is _the_ correct solution when it is actually just _one of many_ correct solutions.

Reply:
Ah! Thanks for that explanation, makes total sense.